server/src/lib/authz-zcap.ts

```diff
@@ -1,94 +1,181 @@
-import { verifyCapabilityInvocation } from "dzcap/invocation-http-signature";
-import type { DID } from "dzcap/did"
+import type { DID } from "dzcap/did";
 import type { IDocumentLoader } from "dzcap/invocation-http-signature";
 import type { ISpace } from "wallet-attached-storage-database/types";
 import type { IZcapCapability } from "dzcap/zcap-invocation-request";
+// @ts-expect-error no types
+import { verifyCapabilityInvocation as dbVerifyCapabilityInvocation } from '@digitalbazaar/http-signature-zcap-verify';
+// @ts-expect-error no types
+import { Ed25519Signature2020 } from '@digitalbazaar/ed25519-signature-2020';
+import { getVerifierForKeyId } from '@did.coop/did-key-ed25519/verifier';
 
-export default async function authorizeWithZcap(request: Request, options: {
-  space(): Promise<ISpace>,
-  expectedTarget?: string,
-  expectedAction?: string,
-  expectedRootCapability?: string | IZcapCapability[]
-  documentLoader?: IDocumentLoader
-  onVerificationError?: (error: unknown) => void
-  resolveRootZcap?: (urn: `urn:zcap:root:${string}`) => Promise<{
-    controller?: DID,
-    "@context": "https://w3id.org/zcap/v1",
-    id: string,
-    invocationTarget: string,
-  }>,
-  // if true, the middleware will respond 401 for any requests with no capability-invocation and signature
-  required?: boolean,
-  trustHeaderXForwardedProto?: boolean
-}): Promise<boolean> {
+function getSpaceRootUrl(url: string): string {
+  const match = url.match(/^(https?:\/\/[^/]+\/space\/[^/]+)/);
+  if (!match) {
+    throw new Error(`Could not determine space root from url: ${url}`);
+  }
+  return match[1];
+}
+
+async function mockVerifyCapabilityInvocation(
+  request: Request,
+  options: {
+    documentLoader: IDocumentLoader;
+    expectedTarget: string;
+    expectedRootCapability: string | IZcapCapability[];
+    expectedAction?: string;
+  }
+) {
+  const u = new URL(request.url);
+  u.protocol = 'https:';
+  const normalizedUrl = u.toString();
+
+  const rt = new URL(options.expectedTarget);
+  rt.protocol = 'https:'; // should already be https
+  // no trailing slash on root
+  if (rt.pathname.endsWith('/')) rt.pathname = rt.pathname.slice(0, -1);
+  const spaceRoot = rt.toString();
+
+  const expectedAction = options.expectedAction || request.method;
+  const verification = await dbVerifyCapabilityInvocation({
+    url: normalizedUrl,
+    method: request.method,
+    suite: new Ed25519Signature2020(),
+    headers: Object.fromEntries(request.headers as any),
+    async getVerifier({ keyId }: { keyId: string }) {
+      if (!keyId) throw new Error('missing keyId');
+      return getVerifierForKeyId(keyId);
+    },
+    documentLoader: options.documentLoader,
+    expectedAction,
+    expectedTarget: normalizedUrl,                     // strict target → will fail on child
+    expectedRootCapability: options.expectedRootCapability,
+    expectedHost: request.headers.get('host') ?? undefined,
```

Comment:
```suggestion
    expectedHost: request.headers.get('host')
```
I don't think that `?? undefined` part is required. headers.get() already returns undefined.